### PR TITLE
Mixed Content linting rule (#20)

### LINKED ISSUE #20: New rule: prevent mixed content
Original issue https://github.com/Elchi3/mdn-doc-tests/issues/143.
Test if sources are HTTPS (e.g. images) to prevent mixed content.

I believe images and things we embed are already https pretty much everywhere, but it looks like we are not actually testing it.

https://twitter.com/DevDocs/status/718123621491847168

### PR DESCRIPTION
**Pull request by MatonAnthony**
Linting rule to prevent mixed content with images as mentionned in the issue #20 

Closes #20

### CODE REVIEW

File: lib/mixed-content.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+const ERROR = require('./doctests.js').ERROR;
+
+// The string must start by https://
+const HTTPS_URL = /^https:\/\//
+
+exports.mixedContent = {
+  name: "mixed_content_name",
+  desc: "mixed_content_desc",
+  check: function mixedContent(rootElement) {
```
Comment (Elchi3) [inline lib/mixed-content.js]: Following the other rules, this should be

`check: function checkMixedContent(rootElement) {`

File: lib/mixed-content.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    let images = rootElement.getElementsByTagName("img");
+    let matches = [];
+
+    for(let index = 0; index < images.length; index++){
+      if(!images[index].getAttribute("href").match(HTTPS_URL)) {
```
Comment (Elchi3) [inline lib/mixed-content.js]: The attribute should be "src" not "href".

File: tests/test-mixed-content.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const assert = require('assert');
+const mixedContent = require('../lib/mixed-content.js').mixedContent;
+const ERROR = require('../lib/doctests.js').ERROR;
+
+describe('mixedContent', function() {
+  it('Should return 2 Errors regarding mixed content', function(done) {
+    const str = '<img href="http://foo.bar/img.png">' +
```
Comment (Elchi3) [inline tests/test-mixed-content.js]: `src` instead of `href` here and below.

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Looks good to me overall. src vs. href needs to be fixed. 
I think for now it is fine to just check img elements. I had a look at https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content but I think we don't allow other elements that could cause mixed content in the first place.

**Comment by MatonAnthony** [comment]
When this PR is merged, it should close #20

**Review by Elchi3 — APPROVED** [review]
r+

**Pull request merged**